src/hooks/useCurrentExecutionStatus.ts

```diff
@@ -0,0 +1,142 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+import { Codec } from '@polkadot/types/types';
+import { useEffect, useState } from 'react';
+
+import { TransactionActionCreators } from '../actions/transactionActions';
+import { SOURCE, TARGET } from '../constants';
+import { useApiTargetPromiseContext } from '../contexts/ApiPromiseTargetContext';
+import { useSourceTarget } from '../contexts/SourceTargetContextProvider';
+import { useTransactionContext, useUpdateTransactionContext } from '../contexts/TransactionContext';
+import useDashboard from '../hooks/useDashboard';
+import useLaneId from '../hooks/useLaneId';
+import useLoadingApi from '../hooks/useLoadingApi';
+import { TransactionStatusEnum } from '../types/transactionTypes';
+import getSubstrateDynamicNames from '../util/getSubstrateDynamicNames';
+
+interface Step {
+  chainType: string;
+  label: string;
+  status: string;
+}
+
+export default function useCurrentExecutionStatus() {
```

Comment:
Why is this a hook instead of being a component? If we want to re-use this for multiple transactions the transaction has to be coming as a parameter.